tools/marvin/marvin/lib/base.py

```diff
@@ -2459,13 +2460,31 @@ def create(cls, apiclient, cluster, services, zoneid=None, podid=None, hyperviso
                   GetDetailExceptionInfo(e)
             return FAILED
 
+    @staticmethod
+    def _check_resource_state(apiclient, hostid, resourcestate):
+        hosts = Host.list(apiclient, id=hostid, listall=True)
+
+        validationresult = validateList(hosts)
+
+        if validationresult[0] == FAIL:
+            raise Exception("Host list validation failed: %s" % validationresult[2])
+
+        if str(hosts[0].resourcestate).lower().decode("string_escape") == str(resourcestate).lower():
+            return True, None
+
+        return False, "Host is not in the following state: " + str(resourcestate)
+
     def delete(self, apiclient):
         """Delete Host"""
         # Host must be in maintenance mode before deletion
         cmd = prepareHostForMaintenance.prepareHostForMaintenanceCmd()
         cmd.id = self.id
         apiclient.prepareHostForMaintenance(cmd)
-        time.sleep(30)
+
+        wait_result, return_val = wait_until(10, 10, Host._check_resource_state, apiclient, self.id, HOST_RS_MAINTENANCE)
```

Comment:
Consider extracting the interval and timeout to constants to make the code more readable.